### PR TITLE
Feat/lesson10

### PR DESCRIPTION
**Pull request by lvwerra**
Review whenever you have time.

### CODE REVIEW

File: notebooks/lesson10_clusters.ipynb [hunk 1]
```diff
@@ -0,0 +1,842 @@
+{
```
Comment (lewtun) [inline notebooks/lesson10_clusters.ipynb:1]: <ul><li>"Learning objectives" for consistency in other lessons. in general i only capitalise the first letter in headings.</li><li>i would write "In this lesson we will have a look at distance and similarity measures to analyse data." (the two applications come later)</li><li>put the "k-nearest neighbour" and "k-means" sentences in a bullet list?</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (lewtun) [inline notebooks/lesson10_clusters.ipynb:1]: <p>I would move the imports and helper functions to be above part 1 and make them level 2 headings <code>## Imports</code>  etc</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (lewtun) [inline notebooks/lesson10_clusters.ipynb:1]: <p>I usually add a block like</p><p> </p><pre class="ql-syntax" spellcheck="false"># uncomment if running on Binder / your laptop
# !pip install --upgrade dslectures
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (lewtun) [inline notebooks/lesson10_clusters.ipynb:1]: <p>migrate to core? also "args" -&gt; "Args"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (lewtun) [inline notebooks/lesson10_clusters.ipynb:1]: <ul><li>"feature vector" -&gt; "feature matrix" for consistency in previous lessons</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (lewtun) [inline notebooks/lesson10_clusters.ipynb:1]: <ul><li>capitalise the docstrings?</li><li>drop the "YOUR CODE" piece or make it explicitly an exercise? sorry just saw the exercise below - maybe it is simpler to define the manhattan distance mathematically as an equation and get them to implement that? the current flow is a bit confusing ...</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (lewtun) [inline notebooks/lesson10_clusters.ipynb:1]: <p>should you tell them how many folds?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Comment (lewtun) [inline notebooks/lesson10_clusters.ipynb:1]: <ul><li>capitalise the docstring?</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lewtun/dslectures/pull/34/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (lvwerra) [reply notebooks/lesson10_clusters.ipynb:1]: <p>I already migrated it, but forgot to delete the original one. good catch!</p>

Comment (lvwerra) [reply notebooks/lesson10_clusters.ipynb:1]: <p>The idea was that i show it to them during the lecture and define it mathematically and then they do it themselves. I added the mathematical definition.</p>

**Pull request merged**